biocrnpyler/parameter.py

```diff
@@ -64,103 +45,319 @@
+
+        for k in parameter_database:
+            if k not in self.parameters or overwrite_parameters:
+                self.parameters[k.parameter_key] = parameter_database[k.parameter_key]
             else:
-                return_field_names[accepted_name] = accepted_field_names[accepted_name][loc_idx]
+                raise ValueError(f"Duplicate parameter detected. Parameter with key = {k} is already in the ParameterDatabase. To Overwrite existing parameters, use overwrite_parameters = True.")
 
-        return return_field_names
+    def load_parameters_from_file(self, filename: str, overwrite_parameters=False) -> None:
+        """Loads parameters from a file to the ParameterDatabase
 
-    @staticmethod
-    def load_parameter_file(filename: str) -> Dict:
-        """load the parameter configuration file into a dictionary
-        :param filename: valid parameter file name
-        :return: a dictionary of the parameters
+        Parameter files must be tab-separated (.tsv or .txt) or comma-separated (.csv) files!
+        :param filename: name of the file (with valid file path)
+        :param overwrite_parameters: whether to overwrite existing entries in the parameter database
         """
-        assert isinstance(filename, str) and len(filename) > 0
-        param_dict = {}
-        # TODO implement search through possible parameter config file locations
-        # Open up the CSV file for reaching
+
+        # Figure out the format of the parameter file from the file extension
         with open(filename) as f:
-            csvreader = csv.DictReader(f, delimiter='\t')
+            file_type = filename.split(".")[-1]
+            if file_type in ["tsv", "txt"]:
+                delimiter = '\t'
+            elif file_type in ["csv"]:
+                delimiter = ","
+            else:
+                raise ValueError("Parameter files must be tab-seperated (.tsv or .txt) or comma-seperated (.csv) files.")
 
-            accepted_field_names = {'mechanism': ['mechanism', 'mechanism_id'],
-                                    'param_name': ["parameter_name", "parameter", "param", "param_name"],
-                                    'part_id': ['part_id', 'part'],
-                                    'param_val': ["val", "value", "param_val", "parameter_value"]
-                                    }
+            csvreader = csv.DictReader(f, delimiter=delimiter)
+            #Used for flexible column headings
+            accepted_field_names = {
+                'mechanism': ['mechanism', 'mechanism_id'],
+                'param_name': ["parameter_name", "parameter", "param", "param_name"],
+                'part_id': ['part_id', 'part'],
+                'param_val': ["val", "value", "param_val", "parameter_value"]
+            }
 
-            field_names = Parameter._get_field_names(csvreader.fieldnames, accepted_field_names)
+            field_names = self._get_field_names(csvreader.fieldnames, accepted_field_names)
```

Comment:
again, is there a reason why this method is called through the object rather than the class?

Reply:
get_fields_names had been a class method before and I kept it that way - in some sense, it isn't linked to the class in any way - it doesn't modify any class variables - so it works to use it as a class method. If you think it should be an instance method though, I have no strong opinions either way.